app/models/ticket_purchase.rb

```diff
@@ -58,4 +60,11 @@ def self.update_quantity(conference, quantity, ticket, user)
     purchase.quantity = quantity if quantity > 0
     purchase
   end
+
+  def pay(payment)
+    update_attributes(paid: true, payment_id: payment)
+    quantity.times do
+      PhysicalTicket.create(ticket_purchase_id: id)
```

Comment:
For creating related objects is better to use the relationship methods it will assign the TicketPurchase object without having to explicitly write it there. 

Also I would prefer to surround this with a transaction. And BTW even better using a one line block style for the iterating part:

```ruby
PhysicalTicket.transaction do
  quantity.times { physical_tickets.create }
end
```